plugins/call_button/call_button.tsx

```diff
@@ -0,0 +1,139 @@
+        return null;
+    }
+
+    const {formatMessage} = props.intl;
+    const [active, setActive] = useState(false);
+    const [clickEnabled, setClickEnabled] = useState(true);
+    const prevSidebarOpen = useRef(props.sidebarOpen);
+
+    useEffect(() => {
+        if (prevSidebarOpen.current && !props.sidebarOpen) {
+            setClickEnabled(false);
+            setTimeout(() => {
+                setClickEnabled(true);
+            }, Constants.CHANNEL_HEADER_BUTTON_DISABLE_TIMEOUT);
+        }
+        prevSidebarOpen.current = props.sidebarOpen;
+    }, [props.sidebarOpen]);
+
+    const style = {
+        container: {
+            marginTop: '16px',
+            height: '32px',
+        } as CSSProperties,
+    };
+
+    if (props.pluginCallComponents.length === 1) {
+        const item = props.pluginCallComponents[0];
+
+        return (
+            <div
+                style={style.container}
+                className='flex-child'
+                onClick={() => {
+                    if (item.action && clickEnabled) {
+                        item.action(props.currentChannel, props.channelMember);
+                    }
+                }}
+                onTouchEnd={() => {
+                    if (item.action && clickEnabled) {
+                        item.action(props.currentChannel, props.channelMember);
+                    }
+                }}
+            >
+                {item.button}
+            </div>
+        );
+    }
+
+    const pluginCallComponents = props.pluginCallComponents.map((item) => {
+        return (
+            <li
+                className='MenuItem'
+                key={item.id}
+                onClick={(e) => {
+                    e.preventDefault();
+                    if (item.action) {
+                        item.action(props.currentChannel, props.channelMember);
+                    }
+                }}
+            >
+
+                {item.dropdownButton}
+            </li>
+        );
+    });
+    return (
+        <div
+            style={style.container}
+            className='flex-child'
+        >
+            <MenuWrapper
+                onToggle={(toggle: boolean) => {
+                    setActive(toggle);
+                }}
+            >
+                <button
+                    type='button'
+                    className={classNames('style--none call-button', 'dropdown', {active})}
+                >
+                    <span
+                        className='icon icon-phone-outline'
+                        aria-label={formatMessage({id: 'generic_icons.call', defaultMessage: 'Call Icon'}).toLowerCase()}
+                    />
+                    <span className='call-button-label'>{'Call'}</span>
+                    <span
+                        className='icon icon-chevron-down'
+                        aria-label={formatMessage({id: 'generic_icons.dropdown', defaultMessage: 'Dropdown Icon'}).toLowerCase()}
+                    />
```

Comment:
it might be easier to use the IconComponents form the compass-icons package here.

Reply:
Would we be able to retain the aria label as well? I guess we could always wrap them but not sure.